### PR TITLE
refactor(2904): clean up apollo federation

### PR DESCRIPTION
**Pull request by ayusham001**
/claim #2904

**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #... _(Replace "..." with the issue number)_

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/core/blueprint/operators/federation.rs [hunk 1]
```diff
@@ -80,24 +80,22 @@ pub fn compile_entity_resolver(inputs: CompileEntityResolver<'_>) -> Valid<IR, S
...
+    let federation_v2_directives = r#"
+        extend schema @link(
+            url: "https://specs.apollo.dev/federation/v2.3",
+            import: [
+                "@key", "@tag", "@shareable", "@inaccessible", 
+                "@override", "@external", "@provides", "@requires", 
+                "@composeDirective", "@interfaceObject"
+            ]
+        )
+    "#;
```
Comment (tusharmath) [inline src/core/blueprint/operators/federation.rs:96]: Don't use strings here. Use `DirectiveDefinition` here and insert it into schema definition.

**Comment by tusharmath** [comment]
Added some comments. Moving to draft to reduce noise and improve CI efficiency. Once you are ready just mark it as "ready to review". Feel free to give a shoutout on the #contributors channel on Discord if you want immediate attention.

**Comment by ayusham001** [comment]
Sure, @tusharmath I have pushed some changes

File: src/core/blueprint/operators/federation.rs [hunk 2]
```diff
@@ -79,27 +79,40 @@ pub fn compile_entity_resolver(inputs: CompileEntityResolver<'_>) -> Valid<IR, S
...
+                    Value::String("@key".to_string()),
+                    Value::String("@tag".to_string()),
+                    Value::String("@shareable".to_string()),
+                    Value::String("@inaccessible".to_string()),
+                    Value::String("@override".to_string()),
+                    Value::String("@external".to_string()),
+                    Value::String("@provides".to_string()),
+                    Value::String("@requires".to_string()),
+                    Value::String("@composeDirective".to_string()),
+                    Value::String("@interfaceObject".to_string()),
```
Comment (tusharmath) [inline src/core/blueprint/operators/federation.rs:104]: better to take a vec of strings and map over them to create a value::string

File: src/core/blueprint/index.rs [hunk 3]
```diff
@@ -261,4 +266,18 @@ mod test {
...
+        let config = include_config!("./fixture/all-constructs.graphql").unwrap();
+        let cfg_module = ConfigModule::from(config);
+
+        let result = compile_service(&cfg_module);
+
+        if let Ok(IR::Service(output)) = result {
+            assert_snapshot!(output);
+        } else {
+            panic!("Expected Valid::Success");
+        }
```
Comment (tusharmath) [inline src/core/blueprint/index.rs:281]: Drop if condition use unwrap.

File: src/core/blueprint/operators/mod.rs [hunk 4]
```diff
@@ -1,4 +1,4 @@
-mod apollo_federation;
+pub mod federation;
```
Comment (tusharmath) [inline src/core/blueprint/operators/mod.rs:1]:
```suggestion
mod federation;
```

File: src/core/blueprint/index.rs [hunk 5]
```diff
@@ -261,4 +266,18 @@ mod test {
...
+
+    #[test]
+    fn test_compile_service_snapshot() {
+        let config = include_config!("./fixture/all-constructs.graphql").unwrap();
+        let cfg_module = ConfigModule::from(config);
+
+        let result = compile_service(&cfg_module);
+
+        if let Ok(IR::Service(output)) = result {
+            assert_snapshot!(output);
```
Comment (tusharmath) [inline src/core/blueprint/index.rs:278]: Snapshot file is missing in the project.

**Pull request closed without merging**